client/allocrunner/networking_bridge_linux.go

```diff
@@ -190,3 +211,49 @@ const nomadCNIConfigTemplate = `{
 	]
 }
 `
+
+const nomadCNIConfigTemplateForTProxy = `{
```

Comment:
My recollection of the original reason why it's a JSON-formatted string and not some kind of CNI config struct is that the `plugin` blocks end up being unstructured `map[string]any`, because each plugin has its own set of accepted fields. So you lose nearly all the type-safety you'd like to get with a config struct.

The original JSON changed once in 5 years (to add the `hairpinMode`), so I think we weren't too concerned about changes when we did this originally as well -- there's just not much in the way of new features to be had here.

I'm inclined to leave it but let me make a quick attempt to add the `consul-cni` block via something like Go templating. Maybe it wouldn't be too terrible to do it that way.

Reply:
Template rendering via `template/text` actually turns out to have a bunch of overhead because of internal mutexes, which we'd only be able to get around by having `sync.Once` constructors for the parsed template and a new struct for the contents. Instead I realized we have a single static block we're looking to add, and we can do that with one extra `%s` at the appropriate location. Not particularly elegant, but it gets the job done very cheaply.